src/model_repository_manager.cc

```diff
@@ -2156,20 +2153,17 @@ ModelRepositoryManager::DependencyGraph::Writeback(
     const std::set<ModelIdentifier>& affected_models)
 {
   for (const auto& model_id : affected_models) {
-    // An affected model can be deleted, so the presence of the model must be
-    // checked.
-    auto* updated_node = updated_dependency_graph.FindNode(
-        model_id, false /* allow_fuzzy_matching */);
-    if (updated_node != nullptr) {
-      auto* node = FindNode(model_id, false /* allow_fuzzy_matching */);
-      // Writeback
-      node->status_ = updated_node->status_;
-      node->checked_ = updated_node->checked_;
-      node->loaded_versions_ = updated_node->loaded_versions_;
-      node->is_locked_ = updated_node->is_locked_;
-      // Notify retry(s)
-      node->retry_notify_cv_->notify_all();
-    }
+    auto* node = GetNode(model_id);
+    auto* updated_node = updated_dependency_graph.GetNode(model_id);
```

Comment:
Do we need to check `if (updated_node)` here for nullptr?

Reply:
Not anymore.

Previously, because models can be removed and the same model may be added during remove, so the graphs could looks like this for example:
```
updated_dependency_graph:    [unmodified_model]
this->dependency_graph_:     [unmodified_model]    [previously_deleted_but_re_added_model]
```
The models contained on the `updated_dependency_graph` is always a subset of `this->dependency_graph_`.

Now, any deleted models are not removed from the dependency graph until they are completely unloaded, so any previously deleted model cannot be re-added until it is fully unloaded (in other words, no more load/unload overlapping). For example,
```
updated_dependency_graph:    [unmodified_model]    [deleted_model]
this->dependency_graph_:     [unmodified_model]    [deleted_model]
```
Thus, the `[previously_deleted_but_re_added_model]` cannot exist, so checking is no longer necessary.

Reply:
Having both of the pointers used here with the function that _can_ return nullptr, even if maybe not in this exact scenario makes me a bit uneasy. 

Do you think we could add some kind of check just in case for future stability if other parts of the code changes that behavior unknowingly?
```cpp
if (!node || !updated_node) {
  LOG_ERROR << "Received unexpected nullptr for dependency graph node: " << model_id << " " << node << " " << updated_node;
} else {
  // Writeback
  ...
}
```

Reply:
The reason why I did not add more checking is this is an unrecoverable error, meaning
```
if (nullptr is returned) {
  log error;
  skip working on nullptr objects;
}
...
```
will corrupt the model repository manager internal states, if allowed to move forward when `nullptr` is returned.

I think we could throw an exception if that happens and terminates the core, instead of a `nullptr` dereference. Given the ordering inside `GetNode()` is specific for the Lock/Unload/Writeback, maybe we can throw it there instead, so it will always return a valid pointer.
```
GetNode(model_id) {
  if (cannot find model from model_id) {
    throw std::invalid_argument("model_id is not on the graph!");
  }
}
```

Reply:
Exception sounds good to me if it shouldn't happen. Better to gracefully throw than hit an unexpected segfault IMO for our own sanity in the future.